### PR TITLE
feat(tools): add sub-agent orchestration (spawn, list, manage)

### PR DESCRIPTION
**Pull request by xdatahubai-a11y**
## Summary

**Problem:** The existing `delegate` tool only supports synchronous sub-agent execution — it blocks the primary agent until the sub-agent completes. This prevents parallel workflows, long-running background tasks, and multi-agent orchestration.

**Why it matters:** Real-world agent workflows need background execution. A primary agent should be able to spawn research, coding, or analysis tasks and continue working while sub-agents run.

**What changed:** Added 3 new tools (`subagent_spawn`, `subagent_list`, `subagent_manage`) and a shared `SubAgentRegistry` for background sub-agent orchestration with full lifecycle management.

### `subagent_spawn`
Spawn a delegate agent in the background via `tokio::spawn`. Returns a `session_id` immediately.
- Respects existing `SecurityPolicy` (autonomy levels, rate limits)
- Enforces `DelegateAgentConfig` depth limits
- Configurable max concurrent sessions (default: 10)
- Uses `CancellationToken` for graceful cancellation

### `subagent_list`
List running/completed/failed/killed sessions with optional status filtering. Read-only.

### `subagent_manage`
Kill running sessions or query status with partial output.

## Validation Evidence

```bash
cargo test --lib tools::subagent
# test result: ok. 56 passed; 0 failed; 0 ignored

cargo check
# Finished dev profile — only pre-existing warnings

cargo clippy 2>&1 | grep subagent
# (no output — clean)
```

## Security Impact

**Risk: Low**
- All write ops (`spawn`, `kill`) enforce `ToolOperation::Act` via `SecurityPolicy`
- Read ops (`list`, `status`) enforce `ToolOperation::Observe`
- `ReadOnly` mode blocks spawn/kill, allows list/status
- Rate limiting enforced per-action
- Concurrent session limit (default: 10) prevents resource exhaustion
- Sub-agents inherit parent security policy and depth limits
- No new network exposure, no credential handling

## Privacy and Data Hygiene

- No PII stored beyond task description
- Sessions auto-cleanup after 1 hour
- No external network calls from registry
- No logging of task content at INFO level

## Rollback Plan

- All changes additive — removing 4 new files + reverting mod.rs/channels/mod.rs restores previous behavior
- No database migrations, no config schema changes
- Existing `delegate` tool completely untouched

## Test Coverage

56 tests across 4 modules: happy paths, error handling, security enforcement, concurrency, parameter validation, edge cases.

### CODE REVIEW

File: src/tools/subagent_manage.rs [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    fn handle_status(&self, session_id: &str) -> anyhow::Result<ToolResult> {
+        // Status is a read operation — no security enforcement needed
+        match self.registry.get_status(session_id) {
+            Some(snap) => {
+                let duration_ms = snap.completed_at.map(|end| {
+                    (end - snap.started_at)
+                        .num_milliseconds()
+                        .max(0)
+                        .cast_unsigned()
+                });
```
[the remark this answers is not in the document]
Comment (xdatahubai-a11y) [reply src/tools/subagent_manage.rs]: Fixed in eaeaa04 — replaced cast_unsigned() with as u64.

File: src/tools/subagent_manage.rs [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+                }
+
+                if let Some(ref r) = snap.result {
+                    output["result"] = json!({
+                        "success": r.success,
+                        "output": if r.output.len() > 500 {
+                            format!("{}... (truncated)", &r.output[..500])
+                        } else {
+                            r.output.clone()
+                        },
```
[the remark this answers is not in the document]
Comment (xdatahubai-a11y) [reply src/tools/subagent_manage.rs:131]: Fixed in eaeaa04 — now uses char_indices().nth(500) to find a safe byte boundary before slicing.

File: src/tools/subagent_registry.rs [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+    pub completed_at: Option<String>,
+    pub duration_ms: Option<u64>,
+}
+
+fn truncate_task(task: &str, max_len: usize) -> String {
+    if task.len() <= max_len {
+        task.to_string()
+    } else {
+        format!("{}...", &task[..max_len])
+    }
```
[the remark this answers is not in the document]
Comment (xdatahubai-a11y) [reply src/tools/subagent_registry.rs:275]: Fixed in eaeaa04 — truncate_task now uses chars().count() for length check and char_indices().nth() for safe slicing. Added truncate_task_multibyte_safe test with emoji input.

File: src/tools/subagent_spawn.rs [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+            agent_name: agent_name_owned.clone(),
+            task: task_owned,
+            status: SubAgentStatus::Running,
+            started_at: Utc::now(),
+            completed_at: None,
+            result: None,
+            handle: None,
+        };
+        self.registry.insert(session);
+
```
[the remark this answers is not in the document]
Comment (xdatahubai-a11y) [reply src/tools/subagent_spawn.rs]: Fixed in eaeaa04 — replaced separate running_count() + insert() with atomic try_insert(session, max) that holds the write lock for both the count check and insertion.

**Comment by chumyin** [comment]
Superseded by #1818 for conflict-free replay on top of latest dev. Continuing with the same CI-green merge policy. Original PR remains open until #1818 merge + green CI, then will be closed by supersede finalization.

**Comment by chumyin** [comment]
Superseded by #1886 for conflict-free replay on top of latest dev. Continuing with the same CI-green merge policy. Original PR remains open until #1886 merge + green CI, then will be closed by supersede finalization.

**Comment by xdatahubai-a11y** [comment]
Closing this PR — all changes have been landed on `dev` by @chumyin via cherry-pick/rebase (commits `067eb8a1`, `90289ccc`, `c90853ba`). An additional fix (`5ac885de`) was added on top by the maintainer.

All 4 CodeRabbit review items were addressed in commit `eaeaa04` before the cherry-pick:
- ✅ `cast_unsigned()` → `as u64`
- ✅ UTF-8 safe truncation (char_indices)
- ✅ Atomic `try_insert` for race condition
- ✅ Docstring coverage

Thanks @chumyin for landing this! PR #1886 can also be closed as superseded.
